packages/common/src/llm/openai.ts

```diff
@@ -259,16 +274,30 @@ function mapToStopReason(
 
 function mapToToolCalls(
   openAIToolCalls: ChatCompletionMessageToolCall[] | undefined,
-  logger: IntegrationLogger
+  logger: IntegrationLogger,
+  provider: string
 ): ToolCall[] | undefined {
   return openAIToolCalls?.reduce((toolCalls, openAIToolCall) => {
     if (openAIToolCall.type === 'function') {
+      let toolCallArguments: ToolCall['function']['arguments']
+
+      try {
+        toolCallArguments = JSON.parse(openAIToolCall.function.arguments)
```

Comment:
note: This was a refactoring to make tool parameters an object rather than as a string for DX purposes.